src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -1147,64 +1093,49 @@ public MetricRegistry getMetricRegistry() {
       return this.metricRegistry;
     }
 
-    private class PartitionBuffer {
+    /**
+     * Implementation of Buffer for Snowpipe based implementation of KC.
+     *
+     * <p>Please note {@link #insert(SinkRecord)} API is called from {@link
+     * com.snowflake.kafka.connector.SnowflakeSinkTask#put(Collection)} API and it is possible the
+     * buffered data is present across multiple PUT apis.
+     *
+     * <p>Check the usage of {@link #getData()} to understand when we would empty this buffer and
+     * when we would generate files in internal stage for snowpipe to ingest later using Snowpipe's
+     * REST APIs
+     */
+    private class SnowpipeBuffer extends PartitionBuffer<String> {
```

Comment:
We should come up with how we want to differentiate between file based snowpipe and snowpipe streaming

Reply:
Yes. We will defined a new PartitionBuffer Type. (More on that in this PR: https://github.com/snowflakedb/snowflake-kafka-connector/pull/382/files)